Classes/common/touchdb/TD_Database.m

```diff
@@ -438,24 +577,43 @@ - (BOOL)close
     _activeReplicators = nil;
 
     [_fmdbQueue close];
+    _fmdbQueue = nil;
+    _keyProviderToOpenDB = nil;
 
     _open = NO;
     _transactionLevel = 0;
+    
     return YES;
 }
 
-- (BOOL)deleteDatabase:(NSError**)outError
+- (BOOL)deleteDatabase:(NSError **)outError
 {
-    CDTLogInfo(CDTDATASTORE_LOG_CONTEXT, @"Deleting %@", _path);
     [[NSNotificationCenter defaultCenter] postNotificationName:TD_DatabaseWillBeDeletedNotification
                                                         object:self];
-    if (_open) {
-        if (![self close]) return NO;
-    } else if (!self.exists) {
-        return YES;
+
+    if (_open && ![self close]) {
```

Comment:
Presuming the code can be called from multiple threads (I've not checked this, but it doesn't look like there's anything preventing this), the database could be closed by another thread between the execution of the check on `_open` and the call to `close`. I'm wondering:

a. Whether this can happen;
b. Whether its worth fixing (it doesn't look like there is potential for data loss, for example).